modules/agar/src/main/ts/ephox/agar/file/PatchInputFiles.ts

```diff
@@ -65,7 +65,11 @@ const cRunOnPatchedFileInput = (files: File[], chain: Chain<any, any>): Chain<an
   cUnpatchInputElement
 ]);
 
+const pRunOnPatchedFileInput = (files: File[], action: () => Promise<void>): Promise<void > =>
+  Chain.toPromise(cRunOnPatchedFileInput(files, Chain.fromPromise(action)))(undefined);
```

Comment:
It really depends on what's the easiest implementation... since we can't use `async`/`await` in McAgar/Agar (since they need to work without the global polyfill) the existing chain is a simpler/cleaner implementation. The rest of the logic uses a mix already.

Reply:
Discussed and we're fine for this now. We'll re-evaluate when we look at deprecating all the old pipeline stuff.